src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/DefaultPageApplicationModelProvider.cs

```diff
@@ -269,10 +275,17 @@ protected virtual PagePropertyModel CreatePropertyModel(PropertyInfo property)
             {
                 // Look for binding info on the handler if nothing is specified on the property.
                 // This allows BindProperty attributes on handlers to apply to properties.
-                var handlerType = property.DeclaringType;
-                var handlerAttributes = handlerType.GetCustomAttributes(inherit: true);
-                var handlerMetadata = _modelMetadataProvider.GetMetadataForType(property.DeclaringType);
-                bindingInfo = BindingInfo.GetBindingInfo(handlerAttributes, handlerMetadata);
```

Comment:
Am I correct this aligns with the PR description?
> Do not allow arbitrary binding attributes to be applied to Razor Page models.

If yes, I'd add we get metadata from a property's container only in unusual cases. Also, have already grabbed all metadata (none of which was binding-related) from the property's type when we reach here.

Either way, probably worth a comment about the special case-y nature of this block.